### PR TITLE
Optimize indexing

### PR DESCRIPTION
**Pull request by loevgaard**
This PR will split the indexing into batches dispatched on the command bus. This will speed up the indexing process proportionally with the number of workers.

### CODE REVIEW

File: src/DataProvider/DefaultDataProvider.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    public function getObjects(string $entity, Index $index): \Generator
+    {
+        $qb = $this
+            ->getManager($entity)
+            ->createQueryBuilder()
+            ->select('o')
+            ->from($entity, 'o');
+
+        $this->eventDispatcher->dispatch(new DataProviderEvent($entity, $index, $qb));
```
Comment (jakubtobiasz) [inline src/DataProvider/DefaultDataProvider.php:34]: I have mixed feelings about the event's name. The event name should reflect a something that happened in the past. So maybe something like `DataObjectProvided` or `PreDataObjectProvided` if you want allow to e.g. modify a $qb?

File: src/Indexer/IndexBuffer.php
Comment (jakubtobiasz) [inline src/Indexer/IndexBuffer.php:1]: Where is it used?

File: src/DataProvider/DefaultDataProvider.php (hunk 1 above)
Comment (loevgaard) [reply src/DataProvider/DefaultDataProvider.php]: Yeah, agreed :)

File: src/Indexer/IndexBuffer.php
Comment (loevgaard) [reply src/Indexer/IndexBuffer.php:1]: In the `DefaultIndexer`

**Pull request merged**